packages/analytics/add-cloud-link-handler/index.ts

```diff
@@ -0,0 +1,46 @@
+import { getProductIntentFromURL } from '../get-product-intent-from-url'
+import { getUTMParamsCaptureState } from '../utm-params-capture'
+
+const destinations: string[] = [
+  'portal.cloud.hashicorp.com',
+  'app.terraform.io',
+]
+const containsDestination = (str: string): boolean =>
+  destinations.some(function (destination) {
+    return str.indexOf(destination) >= 0
+  })
+
+export function addCloudLinkHandler() {
+  if (typeof window === 'undefined') return
+
+  window.addEventListener('click', (event) => {
+    const linkElement = (event.target as HTMLElement).closest('a')
+    if (linkElement && containsDestination(linkElement.href)) {
+      event.preventDefault()
+      const productIntent = getProductIntentFromURL()
+      const forwardedSearchParams: Record<string, string> = {
+        ...getUTMParamsCaptureState(),
+        ...(productIntent && {
+          product: productIntent,
+        }),
+      }
+      try {
+        const url = new URL(linkElement.href)
+        const urlSearchParams = url.searchParams
+        urlSearchParams.forEach((value, key) => {
+          if (!forwardedSearchParams.hasOwnProperty(key)) {
+            forwardedSearchParams[key] = value
+          }
+        })
```

Comment:
@BRKalow @dstaley update since last approved review. This passes along any utm_params inline and if the key matches a param we have in cookies we set that as the value when forwarding.

```
// Example params captured in cookies
const forwardedSearchParams = {
  utm_source: 'dot_com'
}

// Example link with params inline
https://hashicorp.com/procts/terraform?utm_source=cloud_com=utm_medium=twitter

// Expected forwarded url when clicking link to portal
https://portal.cloud.hashicorp.com?utm_source=dot_com&utm_medium=twitter&product=terraform
```

Reply:
Makes sense!